src/core/ext/filters/stateful_session_affinity/stateful_session_affinity_filter.cc

```diff
@@ -0,0 +1,231 @@
+      service_config_call_data->GetMethodParsedConfig(
+          service_config_parser_index_));
+  GPR_ASSERT(method_params != nullptr);
+  auto* cookie_config = method_params->GetConfig(index_);
+  GPR_ASSERT(cookie_config != nullptr);
+  if (!cookie_config->name.has_value()) {
+    return next_promise_factory(std::move(call_args));
+  }
+  // We have a config.
+  // If the config has a path, check to see if it matches the request path.
+  if (cookie_config->path.has_value()) {
+    Slice* path_slice =
+        call_args.client_initial_metadata->get_pointer(HttpPathMetadata());
+    GPR_ASSERT(path_slice != nullptr);
+    absl::string_view path = path_slice->as_string_view();
+    // Matching criteria from
+    // https://www.rfc-editor.org/rfc/rfc6265#section-5.1.4.
+    if (!absl::StartsWith(path, *cookie_config->path) ||
+        (path.size() != cookie_config->path->size() &&
+         cookie_config->path->back() != '/' &&
+         path[cookie_config->path->size() + 1] != '/')) {
+      return next_promise_factory(std::move(call_args));
+    }
+  }
+  // Check to see if we have a host override cookie.
+  auto cookie_value = GetHostOverrideFromCookie(
+      call_args.client_initial_metadata, *cookie_config->name);
+  if (cookie_value.has_value()) {
+    if (GRPC_TRACE_FLAG_ENABLED(grpc_stateful_session_affinity_filter_trace)) {
+      gpr_log(GPR_INFO,
+              "chand=%p: stateful session affinity found cookie %s value %s",
+              this, cookie_config->name->c_str(),
+              std::string(*cookie_value).c_str());
+    }
+    // We have a valid cookie, so add the call attribute to be used by the
+    // xds_override_host LB policy.
+    service_config_call_data->SetCallAttribute(XdsHostOverrideTypeName(),
+                                               *cookie_value);
+  }
+  // Intercept server initial metadata.
+  auto* read_latch = GetContext<Arena>()->New<Latch<ServerMetadata*>>();
+  auto* write_latch =
+      std::exchange(call_args.server_initial_metadata, read_latch);
+  return TryConcurrently(
+             Seq(next_promise_factory(std::move(call_args)),
+                 [cookie_config, cookie_value](ServerMetadataHandle md) {
+                   MaybeUpdateServerInitialMetadata(cookie_config, cookie_value,
+                                                    md.get());
+                   return md;
+                 }))
+      .NecessaryPull(Seq(read_latch->Wait(),
+                         [write_latch, cookie_config,
+                          cookie_value](ServerMetadata** md) -> absl::Status {
+                           if (*md != nullptr) {
+                             MaybeUpdateServerInitialMetadata(
+                                 cookie_config, cookie_value, *md);
+                           }
+                           write_latch->Set(*md);
+                           return absl::OkStatus();
+                         }));
+}
+
+absl::optional<absl::string_view>
```

Comment:
I am always curious about returning string_view from a method. Seems unsafe, especially in the long run as implementation can change unpredictably.